src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourceMappingOptionsControl.xaml.cs

```diff
@@ -0,0 +1,250 @@
+            {
+                //checks to see if keys match
+                if (existingSourceMappings[i] != packageSourceMappings[i])
+                {
+                    return true;
+                }
+                //checks to see if all patterns match
+                if (existingSourceMappings[i].Patterns.Count != packageSourceMappings[i].Patterns.Count)
+                {
+                    return true;
+                }
+                for (int j = 0; j < existingSourceMappings[i].Patterns.Count; ++j)
+                {
+                    if (existingSourceMappings[i].Patterns[j] != packageSourceMappings[i].Patterns[j])
+                    {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        //converts from list of packagesourcemappingsourceItems to a dictonary that can be read by UI
+        private IReadOnlyList<MappingUIDisplay> ReadMappingsFromConfigToUI(IReadOnlyList<PackageSourceMappingSourceItem> originalMappings)
+        {
+            Dictionary<string, List<PackageSourceContextInfo>> uiSourceMappings = new Dictionary<string, List<PackageSourceContextInfo>>();
+            foreach (PackageSourceMappingSourceItem sourceItem in originalMappings)
+            {
+                foreach (PackagePatternItem patternItem in sourceItem.Patterns)
+                {
+                    if (!uiSourceMappings.ContainsKey(patternItem.Pattern))
+                    {
+                        uiSourceMappings[patternItem.Pattern] = new List<PackageSourceContextInfo>();
+                    }
+                    uiSourceMappings[patternItem.Pattern].Add(new PackageSourceContextInfo(sourceItem.Key));
+                }
+            }
+            List<MappingUIDisplay> mappingsCollection = new List<MappingUIDisplay>();
+            foreach (string packageID in uiSourceMappings.Keys)
+            {
+                MappingUIDisplay temp = new MappingUIDisplay(packageID, uiSourceMappings[packageID]);
+                mappingsCollection.Add(temp);
+            }
+            return mappingsCollection.AsReadOnly();
+        }
+
+        //converts from dictonary created by UI to list of packageSourceMappingSourceItems
+        private IReadOnlyList<PackageSourceMappingSourceItem> ReadMappingsFromUIToConfig(ItemsChangeObservableCollection<MappingUIDisplay> uiSourceMappings)
+        {
+            Dictionary<string, List<PackagePatternItem>> mappingsDictonary = new Dictionary<string, List<PackagePatternItem>>();
+            foreach (MappingUIDisplay mappingUIDisplay in uiSourceMappings)
+            {
+                foreach (PackageSourceContextInfo source in mappingUIDisplay.Sources)
+                {
+                    //Contains method did not work since diff instances of MappingUIDisplay even though name is the same
+                    //made own contains method
+                    bool newSource = true;
```

Comment:
I believe line 212-223 can be replaced with, not sure about the above comment.

```
                    if (!mappingsDictonary.Keys.Contains(source.Name))
                    {
                        mappingsDictonary[source.Name] = new List<PackagePatternItem>();
                    }
```

Reply:
Yes thank you